radix-engine/src/blueprints/pool/one_resource_pool/blueprint.rs

```diff
@@ -569,18 +578,20 @@ impl OneResourcePoolBlueprint {
         pool_units_total_supply: Decimal,
         pool_resource_reserves: Decimal,
         pool_resource_divisibility: u8,
-    ) -> Decimal {
+    ) -> Result<Decimal, RuntimeError> {
         let amount_owed = pool_units_to_redeem
```

Comment:
I believe that we should do the division first here since `pool_units_to_redeem` and `pool_resource_reserves` both have an upper bound of `Decimal::MAX` and thus multiplying them first would lead to an overflow if they're large even if the pool units can be claimed.